stage4.py

```diff
@@ -1,48 +1,46 @@
 from card import Card
+from operator import itemgetter
 import random
 
-
+# Adding cards to the deck
 deck = list()
-cards_in_deck = 0
-is_card_identical = 0
-while cards_in_deck < 40:
+is_card_identical = False
+while len(deck) < 40:
     pulled_card = Card()
     for card in deck:
         if card == pulled_card:
-            is_card_identical = 1
+            is_card_identical = True
             break
-    if is_card_identical == 1:
-        is_card_identical = 0
+    if is_card_identical:
+        is_card_identical = False
         continue
     else:
-        cards_in_deck += 1
         deck.append(pulled_card)
 
 random.shuffle(deck)
 
+# - - - - - - - - - - - - - - - - - - - - - - - - -
+
 num_players = 0
-while int(num_players) not in range(2, 7):
-    num_players = input("Enter number of players (between 2 to 6): ")
+while num_players not in range(2, 7):
+    num_players_as_str = input("Enter number of players (between 2 to 6): ")
+    num_players = int(num_players_as_str)
 
-scores = list()
-for i in range(int(num_players)):
-    scores.append(0)
+scores = dict()
```

Comment:
scores is a good name for a list. now when it is a dict it should get a name of mapping like 
players_to_scores
or childeren_to_brother_counts 
or XXX_to_YYY 
you got the point :)